docs/proposals/27-falco-extension.md

```diff
@@ -0,0 +1,318 @@
+      kind: FalcoServiceConfig
+      falcoVersion: <latest Falco version marked as "supported">
+      falcoAutoUpdate: true
+      resources: gardener
+      gardener:
+        useFalcoRules: true
+        useFalcoIncubatingRules: false
+        useFalcoSandboxRules: false
+    type: shoot-falco-service
+```
+
+Falco versions are maintained by the extension in the same way operating
+system-, and Kubernetes versions are maintained by Gardener. If no 
+version is specified the Falco extension will select the currently supported
+version.
+
+There are two methods for managing rules and plugins: either use setup 
+provided by Gardener by setting `resources` to `gardener` or use 
+`falcoctl` by setting `resources` to `falcoctl`.
+
+Custom rules in the Gardener setup can be stored in a ConfigMap and applied 
+to the same project as the cluster. The ConfigMap must be configured as a 
+resource in the cluster as described in [6]:
+
+```
+  resources:
+  - name: customRules
+    resourceRef:
+      apiVersion: v1
+      kind: ConfigMap
+      name: my-custom-rules
+```
+
+This is a template ConfigMap which contains the custom rules. They will be
+deployed and ordered based on the result of the Go programming language
+`strings.Compare` function.
+
+```
+kind: ConfigMap
+  name: my-custom-rules
+  namespace: garden-myproject
+data:
+  myrules.yaml: |-
+    # custom rules go here
+    ...
+```
+
+Events that are generated by Falco are forwarded to Falcosidekick that is 
+deployed alongside Falco. The Falcosidekick webhook output is configured
+to send events to a central location. The Falco extension issues JWT
+tokens containing the cluster identity.
+
+As an alternative users can opt to send events to a custom location by
+defining a `customWebhook` key and configuration. In this case no events 
+will be forwarded to the default event ingestors, events will not be 
+visible in the Gardener dashboard, and there will be no alerting.
+
+#### Falco Versioning
+
+New Falco releases come with new features and might also discontinue existing
+features. Rule sets are usually bound to a particular Falco version. This
+means that updating Falco and/or rules may either break Falco, generate 
+false positive events or might not generate expected events anymore. 
+
+For these reasons we have therefore decided to offer multiple Falco versions
+allow limited control over which version to choose and provide time for 
+necessary updates. We want to roughly follow the shoot versioning policy 
+set out in [7], however with simplifications as the policy has proven to 
+be very complex especially in the operating system area. 
+
+We plan to implement the versioning policy with the following simplifications:
+
+- A linear update path without differentiation of major, minor, and patch 
+  versions.
+- Automated updates always update to the latest supported version.
+- Forced updates always update to the latest supported version.
+- There may be multiple supported versions.
```

Comment:
Good point but I don't have a simple answer right now.

Reply:
If this is future work maybe it is best to mark it as such. In case multi-version support should be in the minimal viable scope there should at least be clarity how a simple update mode should work.

You can also start with just a single version approach and add multi-version support with fancy version update policies later on. However, it would be good to distinguish, which problems are already thought-through and what is still uncertain.